rules/windows/impact_ransomware_file_rename_smb.toml

```diff
@@ -0,0 +1,112 @@
+language = "eql"
+license = "Elastic License v2"
+name = "Suspicious File Renamed via SMB"
+note = """## Triage and analysis
+
+#### Possible investigation steps
+
+- Investigate the source.ip address connecting to port 445 on this host.
+- Identify the user account that performed the file creation via SMB.
+- If the number of files is too high and source.ip connecting over SMB is unusual isolate the host and block the used credentials.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+
+### False positive analysis
+
+- Remote file rename over SMB.
+
+### Related rules
+
+- Third-party Backup Files Deleted via Unexpected Process - 11ea6bec-ebde-4d71-a8e9-784948f8e3e9
+- Volume Shadow Copy Deleted or Resized via VssAdmin - b5ea4bfe-a1b2-421f-9d47-22a75a6f2921
+- Volume Shadow Copy Deletion via PowerShell - d99a037b-c8e2-47a5-97b9-170d076827c4
+- Volume Shadow Copy Deletion via WMIC - dc9c1f74-dac3-48e3-b47f-eb79db358f57
+- Potential Ransomware Note File Dropped via SMB - 02bab13d-fb14-4d7c-b6fe-4a28874d37c5
+
+### Response and remediation
+
+- Initiate the incident response process based on the outcome of the triage.
+- Consider isolating the involved host to prevent destructive behavior, which is commonly associated with this activity.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are identified. Reset passwords for these accounts and other potentially compromised credentials, such as email, business systems, and web services.
+- If any other destructive action was identified on the host, it is recommended to prioritize the investigation and look for ransomware preparation and execution activities.
+- If any backups were affected:
+  - Perform data recovery locally or restore the backups from replicated copies (cloud, other servers, etc.).
+- Determine the initial vector abused by the attacker and take action to prevent reinfection through the same vector.
+- Using the incident response data, update logging and audit policies to improve the mean time to detect (MTTD) and the mean time to respond (MTTR).
+"""
+risk_score = 73
+rule_id = "78e9b5d5-7c07-40a7-a591-3dbbf464c386"
+setup = """## Setup
+
+If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2,
+events will not define `event.ingested` and default fallback for EQL rules was not added until version 8.2.
+Hence for this rule to work effectively, users will need to add a custom ingest pipeline to populate
+`event.ingested` to @timestamp.
+For more details on adding a custom ingest pipeline refer - https://www.elastic.co/guide/en/fleet/current/data-streams-pipeline-tutorial.html
+"""
+severity = "high"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Impact", "Resources: Investigation Guide", "Data Source: Elastic Defend"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
```

Comment:
I think there are a few concerns here that we may be able to work out (or at least warn against):
1. The first subquery will potentially match on a lot, holding on to a lot of state and consuming resources (internally searched shows ~25k events/30 days for a mid size deployment)
2. since it is only joined by the `host.id`, a high likelihood race condition exists (especially if it is a server or system with concurrent users)
3. 3 runs doesn't seem like enough, but more means more resource consumption


Some thoughts to solve
1. tighten maxspan (but also increases FN)
2. we drop the SMB check altogether and use threshold or ESQL to find a high amount of renamed files by pid 4


Or a radical alternative
1. BBR on incoming smb - 1m interval, 2m lookback
2. ESQL BBR for large aggregate file renames - 1m interval, 2m lookback
3. sequence rule on BBR1 then BBR2 based on event.ingested - 5/9

🤷 don't feel great about the last one, but it _could_ reliably work

I feel like our best option is:

>2. we drop the SMB check altogether and use threshold or ESQL to find a high amount of renamed files by pid 4

Or, lastly, we add disclaimers that this is potentially costly and FP prone

Reply:
> since it is only joined by the host.id, a high likelihood race condition exists (especially if it is a server or system with concurrent users)

the effect of an unrelated smb network connection interfering with the sequence is decreased by `process.pid == 4 and user.id : ("S-1-5-21*", "S-1-12-*")` (pid 4 is normally running as s-1-5-18, when not it means its impersonating a user which matches with SMB activity).

> Or, lastly, we add disclaimers that this is potentially costly and FP prone

I think I will go with a disclaimer for performance impact (the existing seq provides an alert with the source.ip and user.name) and add an extra threshold rule for high freq of file renames by pid 4 (users who don't want to go for the seq one can enable the threshold one).

Reply:
++ SGTM, thanks @Samirbous 👍